components/brave_referrals/browser/brave_referrals_service.cc

```diff
@@ -252,6 +256,15 @@ void BraveReferralsService::OnReferralInitLoadComplete(
                << ", response code: " << response_code
                << ", payload: " << safe_response_body
                << ", url: " << referral_init_loader_->GetFinalURL().spec();
+    if (!initialization_timer_) {
+      initialization_timer_ = std::make_unique<base::RepeatingTimer>();
```

Comment:
I think OneShotTimer is more fits here.
Whenever reach here, I think we just need to start new one shot timer.

Reply:
seems reasonable